associations.rb

```diff
@@ -46,26 +50,38 @@ def generate_migrations
 
 
 class Hotel < ActiveRecord::Base
-  #insert our associations here
+  has_many :rooms
+  has_many :bookings, :through => :rooms
  
   def to_s
     "#{name} with #{rooms.count} rooms"
   end
+
+  def booked_guests
+    [].tap do |r|
+      self.bookings.each { |b| r << b.guest }
+    end
```

Comment:
I think [ActiveRecord's `has_many :through` association](http://guides.rubyonrails.org/association_basics.html#the-has-many-through-association) can take care of this for you.